src/renderer/pathTracer.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2021 F. Lotfi & D. Kane All rights reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file.
+ */
+
+#ifndef __PATHTRACER_H__
+#define __PATHTRACER_H__
+
+#include <functional>
+#include <glm/glm.hpp>
+#include <math.h>
+#include <memory>
+
+#include "raycast/ray.h"
+#include "renderer/pixelBuffer.h"
+#include "scene/scene.h"
+
+class PathTracer {
+  typedef std::function<void()> drawCallBack;
+
+private:
+  std::unique_ptr<Scene> mScene;
+  const float mAspectRatio;
+  PixelBuffer *mPixelBuffer;
+  void draw();
+
+  glm::vec3 colorPixel(const unsigned int i, const unsigned int j);
+  glm::vec3 colorRay(const Ray &ray);
+
+public:
+  PathTracer();
+
+  void renderScene();
+
+  void setPixelBuffer(PixelBuffer *pixelBuffer) { mPixelBuffer = pixelBuffer; };
+
+  void setScene(Scene &scene) { mScene = std::make_unique<Scene>(scene); }
```

Comment:
this api is going to come back to bite us in the ass one day. Right now you have it where we are taking a stack allocated object in the main func. We can't free memory like that so wrapping in a unque pointer is going to blow up one day if we move the code around.  make this take a unquePtr and then do a std::move to get it into mScene.